cli/client/cosmos.go

```diff
@@ -68,16 +75,16 @@ func createCosmosHTTPJSONRequest(method, urlPath, jsonPayload string) *http.Requ
 
 func createCosmosURL(urlPath string) *url.URL {
 	// Try to fetch the Cosmos URL from the system configuration
-	if len(config.CosmosUrl) == 0 {
-		config.CosmosUrl = OptionalCLIConfigValue("package.cosmos_url")
+	if len(config.CosmosURL) == 0 {
+		config.CosmosURL = OptionalCLIConfigValue("package.cosmos_url")
 	}
 
 	// Use Cosmos URL if we have it specified
-	if len(config.CosmosUrl) > 0 {
+	if len(config.CosmosURL) > 0 {
```

Comment:
I'm not sure I follow. The first conditional checks to see if we have one specified - if we haven't, then it'll attempt to read it from the DC/OS CLI's configuration. This is similar to how http.go does it. The reason we don't always set it is that the user may manually override it (in fact, we do this for unit testing).

The second conditional uses that URL if we have it specified - I can't see of a clean way to merge this into the previous conditional that wouldn't be more confusing than the current implemtnation.